### ISSUE TITLE
Process C:\my-project not running

### ISSUE THREAD

**Issue by codygman** [syl20bnr/spacemacs#3443]
Steps to reproduce using Windows 8.1 and (spacemacs-0.104.2):

1. mkdir sample-project; cd sample-project; cabal init
2. select executable when asked whether executable or library
3. write `main = undefined` in a file named Main.hs
4. `SPC m s b`
5. Get error message `C:\my project not running`

**Comment by PierreR** [comment]
I am not an windows (and I don't recognize the error message) but just in case could you first do `SPC m s s` (or `,ss`) before loading the file with (`,sb`)

**Comment by codygman** [comment]
Upon verifying my repro steps above I notice there is an error upon opening:

```
File mode specification error: (error "Process c:\\sample-project not running")
```

It opens up the shell window and I can confirm it works:
```
Hello, Haskell!
If I break, you can:
  1. Restart:           M-x haskell-process-restart
  2. Configure logging: C-h v haskell-process-log (useful for debugging)
  3. General config:    M-x customize-mode
  4. Hide these tips:   C-h v haskell-process-show-debug-tips
λ> 1 + 1
2
λ> 
```

Then I go back to the file and `,sb` and it tells me `Process C:\sample-project not running`.

**Comment by codygman** [comment]
For some odd reason it seems that debugging made the issue disappear. I tried again without debug-on-entry on and the bug still exists.

I've also verified that haskell-mode on vanilla emacs works just fine.

**Comment by codygman** [comment]
Upon opening a Haskell file I get:

```
File mode specification error: (error "Process c:\\sample-project not running")
```

After setting debug-on-entry and reopening a Haskell file I get this backtrace:

```
Debugger entered--Lisp error: (error "Process c:\\sample-project not running")
  process-send-string(#<process c:\sample-project> "check c:/sample-project/Main.hs\n")
  ghc-with-process("check c:/sample-project/Main.hs\n" ghc-check-callback #[nil "\301\211\207" [mode-line-process " -:-"] 2])
  ghc-check-syntax()
  ghc-init()
  run-hooks(change-major-mode-after-body-hook prog-mode-hook haskell-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook prog-mode-hook haskell-mode-hook))
  run-mode-hooks(haskell-mode-hook)
  haskell-mode()
  set-auto-mode-0(haskell-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer Main.hs> "c:/sample-project/Main.hs" nil nil "c:/sample-project/Main.hs" (2533274790702288 3267088014))
  find-file-noselect("c:/sample-project/Main.hs" nil nil nil)
  find-file("c:/sample-project/Main.hs")
  find-file-existing("c:/sample-project/Main.hs")
  (lambda (&rest ignore) (find-file-existing "c:/sample-project/Main.hs"))((push-button :args nil :value "c:/sample-project/Main.hs" :action (lambda (&rest ignore) (find-file-existing "c:/sample-project/Main.hs")) :mouse-face highlight :follow-link "
" :button-prefix "" :button-suffix "" :format "%[%t%]" :button-overlay #<overlay from 4027 to 4052 in *spacemacs*> :from #<marker (moves after insertion) at 4027 in *spacemacs*> :to #<marker at 4052 in *spacemacs*>) nil)
  widget-apply((push-button :args nil :value "c:/sample-project/Main.hs" :action (lambda (&rest ignore) (find-file-existing "c:/sample-project/Main.hs")) :mouse-face highlight :follow-link "
" :button-prefix "" :button-suffix "" :format "%[%t%]" :button-overlay #<overlay from 4027 to 4052 in *spacemacs*> :from #<marker (moves after insertion) at 4027 in *spacemacs*> :to #<marker at 4052 in *spacemacs*>) :action nil)
  widget-apply-action((push-button :args nil :value "c:/sample-project/Main.hs" :action (lambda (&rest ignore) (find-file-existing "c:/sample-project/Main.hs")) :mouse-face highlight :follow-link "
" :button-prefix "" :button-suffix "" :format "%[%t%]" :button-overlay #<overlay from 4027 to 4052 in *spacemacs*> :from #<marker (moves after insertion) at 4027 in *spacemacs*> :to #<marker at 4052 in *spacemacs*>) nil)
  widget-button-press(4027)
  evil-ret-gen(nil nil)
  evil-ret(nil)
  funcall-interactively(evil-ret nil)
  call-interactively(evil-ret nil nil)
  command-execute(evil-ret)
```

**Comment by codygman** [comment]
This backtrace confuses me since I tried vanilla emacs, adding melpa, then installing haskell-mode and things seem to work fine. So something with spacemacs is messing up the process-send-string call?

**Comment by codygman** [comment]
New to trying to debug elisp, but it looks like the process is dying before the invocation of process-send-string?

**Comment by codygman** [comment]
This error only happens when ghc-mod isn't properly installed. I believe it has something to do with the [ghc-with-process](https://github.com/kazu-yamamoto/ghc-mod/blob/master/elisp/ghc-process.el#L46) function where possibly ghc-process-process-name is returning different results and instead that call on line 46 should use the previously bound variable name.

I tried changing it then deleting *.elc files in the ghc directory, but that might not be enough. I remember having to generate *.elc files for haskell mode but I'm not sure how to do it manually for ghc.

**Comment by d12frosted** [comment]
You can always set `haskell-enable-ghc-mod` to `nil`. But, you probably should report this problem in `ghc-mod` issue tracker.

**Comment by d12frosted** [comment]
Oh, actually there were some problems with disabling `ghc-mod-supoort`, should be fixed by #3543.

**Comment by TheBB** [comment]
Glad to see you fixed it. When #3543 is fixed it should be possible to disable `ghc-mod` support without issues. Closing this on the recommendation of @d12frosted.
